translators/DirectionalLight.cpp

```diff
@@ -0,0 +1,310 @@
+//
+// Copyright 2017 Animal Logic
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.//
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "DirectionalLight.h"
+#include "pxr/usd/usdLux/distantLight.h"
+#include "pxr/usd/usdGeom/xformCache.h"
+
+#include "AL/usdmaya/utils/DgNodeHelper.h"
+#include "AL/usdmaya/fileio/AnimationTranslator.h"
+#include "AL/usdmaya/fileio/translators/DgNodeTranslator.h"
+
+#include "maya/MAngle.h"
+#include "maya/MFnDirectionalLight.h"
+#include "maya/MFnNumericAttribute.h"
+#include "maya/MNodeClass.h"
+
+namespace AL {
+namespace usdmaya {
+namespace fileio {
+namespace translators {
+
+AL_USDMAYA_DEFINE_TRANSLATOR(DirectionalLight, PXR_NS::UsdLuxDistantLight)
+
+//----------------------------------------------------------------------------------------------------------------------
+MObject DirectionalLight::m_pointWorld;
+MObject DirectionalLight::m_lightAngle;
+MObject DirectionalLight::m_color;
+MObject DirectionalLight::m_intensity;
+MObject DirectionalLight::m_exposure;
+MObject DirectionalLight::m_diffuse;
+MObject DirectionalLight::m_specular;
+MObject DirectionalLight::m_normalize;
+MObject DirectionalLight::m_enableColorTemperature;
+
+//----------------------------------------------------------------------------------------------------------------------
+MStatus DirectionalLight::initialize()
+{
+    MStatus status = MS::kSuccess;
```

Comment:
The use of status in this method is a little strange - whilst it's passed to all methods that return it - the only check exists at the very end of the method. This has the potential to hide a bunch of errors.